drahtbot/src/features/summary_comment.rs

```diff
@@ -0,0 +1,493 @@
+use crate::errors::DrahtBotError;
+use crate::errors::Result;
+use crate::Context;
+use crate::GitHubEvent;
+use crate::Repository;
+use async_trait::async_trait;
+use futures_util::TryStreamExt;
+use lazy_static::lazy_static;
+
+pub struct SummaryCommentFeature {
+    meta: FeatureMeta,
+}
+
+impl SummaryCommentFeature {
+    pub fn new() -> Self {
+        Self {
+            meta: FeatureMeta::new(
+                "Summary Comment",
+                "Creates a summary comment on pulls requests with an ACK tracker.",
+                vec![
+                    GitHubEvent::PullRequest,
+                    GitHubEvent::PullRequestReview,
+                    GitHubEvent::IssueComment,
+                ],
+            ),
+        }
+    }
+}
+
+#[async_trait]
+impl Feature for SummaryCommentFeature {
+    fn meta(&self) -> &FeatureMeta {
+        &self.meta
+    }
+
+    async fn handle(
+        &self,
+        ctx: &Context,
+        event: GitHubEvent,
+        payload: &serde_json::Value,
+    ) -> Result<()> {
+        let action = payload["action"]
+            .as_str()
+            .ok_or(DrahtBotError::UnknownEvent)?;
+
+        let repo_user = payload["repository"]["owner"]["login"]
+            .as_str()
+            .ok_or(DrahtBotError::KeyNotFound)?;
+
+        let repo_name = payload["repository"]["name"]
+            .as_str()
+            .ok_or(DrahtBotError::KeyNotFound)?;
+
+        let repo = Repository {
+            owner: repo_user.to_string(),
+            name: repo_name.to_string(),
+        };
+
+        println!("Handling event: {:?}", event);
+        println!("Action: {:?}", payload["action"]);
+        match event {
+            GitHubEvent::PullRequest if action == "opened" => {
+                create_summary_comment(payload, &ctx.octocrab).await?
+            }
+
+            GitHubEvent::PullRequest if action == "synchronize" => {
+                let pr_number = payload["number"]
+                    .as_u64()
+                    .ok_or(DrahtBotError::KeyNotFound)?;
+
+                refresh_summary_comment(ctx, repo, pr_number).await?
+            }
+            GitHubEvent::IssueComment if payload["issue"].get("pull_request").is_some() => {
+                println!("Issue comment on pull request");
+                let pr_number = payload["issue"]["number"]
+                    .as_u64()
+                    .ok_or(DrahtBotError::KeyNotFound)?;
+                refresh_summary_comment(ctx, repo, pr_number).await?
+            }
+            GitHubEvent::PullRequestReview => {
+                let pr_number = payload["pull_request"]["number"]
+                    .as_u64()
+                    .ok_or(DrahtBotError::KeyNotFound)?;
+                refresh_summary_comment(ctx, repo, pr_number).await?
+            }
+            _ => {}
+        }
+        Ok(())
+    }
+}
+
+fn summary_comment_template(initial: bool, acks: Option<Vec<Review>>) -> String {
+    let mut comment = r#"## Summary
```

Comment:
Given that a synchronous update is impossible, it might be better to re-use the existing "async" comment structure. See https://github.com/MarcoFalke/DrahtBot/blob/97274a4dcd00a207d4e6906e4b8272b7493a3356/scripts/util/util.py#L15

(I'll port it to rust soon)

Otherwise every pull request might have two summary comments.

Reply:
> Given that a synchronous update is impossible, it might be better to re-use the existing "async" comment structure. See 

Right good idea.

> This could also allow to get rid of having to pass the name of the bot via command line?

Checking the bot username could make sure that a user doesn't include this hidden comment to mess with the bot no? 
Or the username could be deduced at runtime with an API request.

Reply:
Would be hard to mess with the bot, as the first comment is picked. Even if a user races the bot, it doesn't seem to be able to cause any harm?

Reply:
This would also make it easier to replace the bot if a bot user account gets blocked/deleted

Reply:
> (I'll port it to rust soon)

Done.

You should now be able to call `let cmt = util::get_metadata_sections(pull_number)` and then `util::update_metadata_comment(cmt, "reviews ...", util::IdComment::SecReviews)`

Reply:
I addressed your change (had to rebase), seems to work great!
If the summary comment was not already created on a PR, it will be created but won't appear at the top. This should be fine for old PRs.